### ISSUE TITLE
Map not focusable

### ISSUE THREAD

**Issue by Sebubu** [openlayers/openlayers#10109]
**Describe the bug**
The openlayers map is not focusable with a normal mouse click anymore. We are using OpenLayers as a form input to draw polygons. Therefore the map needs to be focusable. 

Usually that worked by just putting `tabindex="0"` on the map div but as we upgraded from ol 4 to ol 6.0.1 this doesn't work anymore.

I guess the focus event is caught somewhere and not bubbled up anymore to the main div.

**To Reproduce**
1. Go to https://stackblitz.com/edit/js-tx81gf
2. Left click on the map -> No focus
3. Right click on the map -> Focus

**Expected behavior**
The map should be focusable even with a normal left click.

I have searched through the code but I haven't found a line where the focus event is caught.

Best regards
Sevi

**Comment by tonio** [comment]
You can have this behavior this a bit of css:
```css
#map:not(:focus) .ol-viewport {
  pointer-events: none;
}
```

**Comment by Sebubu** [comment]
> You can have this behavior with a bit of css:
> 
> ```css
> #map:not(:focus) .ol-viewport {
>   pointer-events: none;
> }
> ```

Thanks, but this breaks the ol interactions like `Select`.

I finally want to subscribe to keydown events on the map. For example I remove a polygon if the delete key is pressed. Maybe there is a way to subscribe to keyevents in ol internally?

**Comment by Sebubu** [comment]
The ol internal keydown event doesn't work either. I have extend the stackblitz app above.

```javascript
map.on('keydown', (event) => {
  console.log('keydown', event);
});
```

**Comment by Sebubu** [comment]
I found the culprit.

The `DragDown` interaction returns true in [handleDownEvent](https://github.com/openlayers/openlayers/blob/7be483514442bc5518274f6ebfea7501dad68ae2/src/ol/interaction/DragPan.js#L162) and therefore the `Pointer` interaction calls [preventDefault()](https://github.com/openlayers/openlayers/blob/1f5dfef2006defbb2b60adca612c3dfd3368f281/src/ol/interaction/Pointer.js#L138) on the click event which prevents the focus. 

Don't know how to fix this problem but it breaks pretty all interactions with key events like the [KeyboardZoom](https://openlayers.org/en/latest/apidoc/module-ol_interaction_KeyboardZoom.html) and [KeyboardPan](https://openlayers.org/en/latest/apidoc/module-ol_interaction_KeyboardPan-KeyboardPan.html).

#### Workaround
Removing the DragPan (extends Pointer) from the default interactions makes the map focusable.

**Comment by cesboa** [comment]
I think it's the same issue on touch. On iPad with iOS 12 OL6 has some wired focus issues. When using a CesiumJS view side by side Cesium can't gain control, while when downgrading OL5.3.3 (and only that) the touch/input works like a charm. Or is this totally unrelated?

**Comment by Sebubu** [comment]
> I think it's the same issue on touch. On iPad with iOS 12 OL6 has some wired focus issues. When using a CesiumJS view side by side Cesium can't gain control, while when downgrading OL5.3.3 (and only that) the touch/input works like a charm. Or is this totally unrelated?

@cesboa I have included a [workaround](https://stackblitz.com/edit/js-tx81gf) in the example. If this fixes your problem it is likely related.

**Comment by mike-000** [comment]
KeyboardZoom and KeyboardPan are not working on a full page map with
 `keyboardEventTarget: document` 
set so it seems more than a focusing problem.

**Comment by Sebubu** [comment]
Key events are only sent by the element which has been focused before. See https://stackoverflow.com/questions/10214547/do-browser-key-events-fire-bubble-only-to-elements-in-focus

Lack of focus -> no key events -> no KeyboardZoom and KeyboardPan

**Comment by mike-000** [comment]
With a full page map and using `keyboardEventTarget: document` the difference between OL5 and OL6 appears to be KeyboardZoom and KeyboardPan do not work if a control has focus in OL6 while they do work in OL5 -  so clicking on a control button stops keyboard interactions, and you must use the tab key to move focus off the controls.

**Comment by ahocevar** [comment]
Maybe some of the issues described here are fixed with #10187.

**Comment by Sebubu** [comment]
> Maybe some of the issues described here are fixed with #10187.

Interesting. Both examples in #10187 work on my computer. The option `{ onFocusOnly: true }` seems to fix the focus issue.

**Comment by ahocevar** [comment]
> Interesting. Both examples in #10187 work on my computer. The option { onFocusOnly: true } seems to fix the focus issue.

That's only broken on touch devices. Computers with mouse/trackpad work fine.
